core/clustersmngr/middleware_test.go

```diff
@@ -0,0 +1,69 @@
+package clustersmngr_test
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/weaveworks/weave-gitops/core/clustersmngr"
+	"github.com/weaveworks/weave-gitops/core/clustersmngr/clustersmngrfakes"
+	"github.com/weaveworks/weave-gitops/pkg/server/auth"
+)
+
+func TestWithClustersClientsMiddleware(t *testing.T) {
+	cluster := makeLeafCluster(t)
+	clustersFetcher := &clustersmngrfakes.FakeClusterFetcher{}
+	clustersFetcher.FetchReturns([]clustersmngr.Cluster{cluster}, nil)
+
+	g := NewGomegaWithT(t)
+
+	defaultHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+	middleware := func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			clientsPool := clustersmngr.ClientsPoolFromCtx(r.Context())
+
+			clients := clientsPool.Clients()
+			if _, ok := clients[cluster.Name]; !ok {
+				g.Fail("leaf cluster client not present")
+			}
```

Comment:
we can simplify to this if you like
```suggestion
			g.Expect(clientsPool.Clients()).To(HaveKey(cluster.Name))
```